subset_vds.py

```
@@ -0,0 +1,340 @@
+
+    if n_samples:
+        subset_sample_list = get_subset_sample_list(input_vds, n_samples)
+        if intervals:
+            parsed_intervals = parse_intervals(intervals)
+            parsed_locus = convert_intervals_to_locus(parsed_intervals)
+            subset_vds = subset_by_locus(
+                parsed_locus,
+                subset_by_samples(input_vds, subset_sample_list),
+            )
+        else:
+            subset_vds = subset_by_samples(input_vds, subset_sample_list)
+    elif intervals:
+        parsed_intervals = parse_intervals(intervals)
+        parsed_locus = convert_intervals_to_locus(parsed_intervals)
+        subset_vds = subset_by_locus(parsed_locus, input_vds)
+
+    write_outputs(output_formats, subset_vds, subset_sample_list, infile_name)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="VDS subsetting script")
+    parser.add_argument("--vds-path", help="Path to VDS in GCP.", required=True)
+    parser.add_argument(
+        "--n-samples",
+        help="Number of samples to subset the VDS down to.",
+        required=False,
+        type=int,
+    )
+    parser.add_argument(
+        "--intervals",
+        help="Interval(s) provided either on the command line in the format chr:start-end or in a text file, one per line to keep in the VDS.",
+        required=False,
+    )
+    parser.add_argument(
+        "--output-formats",
+        help="Comma separated string indicating what output formats you would like. One of [vcf, bed, vds]",
+        required=True,
+        nargs="+",
+        choices=["vcf", "bed", "vds"],
+    )
+    parser.add_argument(
+        "--random-seed",
+        help="Seed for the random number generator when subsetting by samples",
+        required=False,
+        default=19700101,
+        type=int,
+    )
+    args, failures = parser.parse_known_args()
+    if failures:
+        parser.print_help()
+        raise AttributeError(f"Invalid arguments {failures}")
```

It's not important here, but I completely misunderstood this method, and was using it incorrectly.

`parser.parse_args()` attempts to parse all arguments, and will fail if there are additional arguments provided.

`parser.parse_known_args()` only parses the listed arguments, and will tolerate other arguments being present (it'll return the other un-digested ones). I'm not sure when you'd want to parse only a subset of the CLI arguments, but that would be permitted in parse_known_args. 

TL;DR - if you always want to fail when unknown arguments are present, the standard parse_args is the way to go

---

Ah, I had assumed that the functionality was reversed, as I'd seen it used around CPG like that. I'll make that change

---

Almost certainly me writing it/others copying my mistake 😆